### ISSUE TITLE
Update to kotlin 1.4.0

### ISSUE THREAD

**Issue by NatsuOnFire** [Kodein-Framework/Kodein-DI#318]
New version of kotlin (1.4.0) released.

When I use it with **kodein-di-jvm 7.0.0** I have these warnings appearing:

```
[WARNING] Runtime JAR files in the classpath should have the same version. These files were found in the classpath:
    ../.m2/repository/org/jetbrains/kotlin/kotlin-stdlib/1.4.0/kotlin-stdlib-1.4.0.jar (version 1.4)
    ../.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-common/1.4.0/kotlin-stdlib-common-1.4.0.jar (version 1.4)
    ../.m2/repository/org/jetbrains/kotlin/kotlin-reflect/1.4.0/kotlin-reflect-1.4.0.jar (version 1.4)
    ../.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-jdk7/1.3.72/kotlin-stdlib-jdk7-1.3.72.jar (version 1.3)
    ../.m2/repository/org/jetbrains/kotlin/kotlin-stdlib-jdk8/1.3.71/kotlin-stdlib-jdk8-1.3.71.jar (version 1.3)
[WARNING] Some runtime JAR files in the classpath have an incompatible version. Consider removing them from the classpath
```

**Comment by romainbsl** [comment]
Kotlin 1.4 has just been announced. We are working on the upgrade with some major fixes. We'll keep you posted when it's up!

**Comment by NatsuOnFire** [comment]
It's great, thank you for the responsiveness ^_^

**Comment by romainbsl** [comment]
While waiting you can use our dev channel 

https://dl.bintray.com/kodein-framework/kodein-dev


There is a version `7.1.0-master-80`. Already working with kotlin 1.4.0

**Comment by supertote** [comment]
It would be nice that when migrating to Kotlin 1.4.0 you also change the production of the JS part so that the generated js artifacts contain both the LEGACY and the IR variants. This is the only thing which prevents me from using the IR backend which produces much tighter js code.
I have not tried de 7.1.0-master-80 but at least the 7.1.0-kotlin-1.4-rc-85 was only containing the LEGACY artifacts.

**Comment by dalewking** [comment]
Any update on this? I think this is the last obstacle for moving to 1.4 for our project

**Comment by romainbsl** [comment]
Kotlin 1.4 is still usable from our dev channel `7.1.0-master-88`. 
7.1 release will be there soon.

**Comment by xajik** [comment]
Hey guys, just want to confirm that this issue is because of 1.4 migration? 
`> Task :app:compileKotlinIosX64
w: Skipping "/Users/user/.gradle/caches/modules-2/files-2.1/org.kodein.di/kodein-di-iosx64/7.0.0/ccc9ef9022eb8cf1593b92bd2c3c17bea317c157/kodein-di.klib" as it is a pre 1.4 library
e: Could not find "/Users/user/.gradle/caches/modules-2/files-2.1/org.kodein.di/kodein-di-iosx64/7.0.0/ccc9ef9022eb8cf1593b92bd2c3c17bea317c157/kodein-di.klib" in [/Users/user/Project/application/app/applicationandroid/application, /Users/user/.konan/klib, /Users/user/.konan/kotlin-native-prebuilt-macos-1.4/klib/common, /Users/user/.konan/kotlin-native-prebuilt-macos-1.4/klib/platform/ios_x64].
5:05:48 PM: Task execution finished ':app:packForXcode'.`

**Comment by turansky** [comment]
@xajik yes, you use incompatible Kodein version

**Comment by romainbsl** [comment]
stable version `7.1.0` is available with kotlin 1.4.0.

**Comment by romainbsl** [comment]
Yes because starting with 1.4.0 we are no longer forced to be aligned on Kotlin versions for every dependencies.

Kotlin 1.4.10 only brings compiler fixes that we didn't need in Kodein libs.
